framework/graphics/vulkan_struct_deep_copy.h

```diff
@@ -48,14 +48,14 @@ template <typename T>
 size_t vulkan_struct_deep_copy(const T* structs, uint32_t count, uint8_t* out_data);
 
 /**
- * @brief   vulkan_struct_deep_copy_pnext is similar to 'vulkan_struct_deep_copy',
+ * @brief   vulkan_struct_deep_copy_stype is similar to 'vulkan_struct_deep_copy',
  *          with additional type-resolution based on sType-member.
  *
  * @param   pNext       a pointer to a pNext-chain
  * @param   out_data    pointer to an output-array or nullptr
  * @return  number of bytes required for deep-copy
  */
-size_t vulkan_struct_deep_copy_pnext(const void* pNext, uint8_t* out_data);
+size_t vulkan_struct_deep_copy_stype(const void* pNext, uint8_t* out_data);
```

Comment:
Why the name change? It still takes `pNext` as a parameter after all.

Reply:
sole reason I renamed, was to emphasize the fact that it 'can' be in fact used for any vulkan-structure. it's not so much about whether it's used as pNExt or not, but rather that the type will be resolved based on its `sType` member.

Reply:
there are few union-types where this can be useful, namely `VkAccelerationStructureGeometryDataKHR`. you 'could' just pass a void* to that union

Reply:
Makes sense. Thank you.

Reply:
but since it's 'mostly' used for pNext chains, I kept the param-name and doc :)